pinc/theme.inc

```diff
@@ -230,29 +299,59 @@ function html_navbar()
         $links = [];
         $links[] = ['text' => $login_form];
         $links[] = ['text' => _('Register'), 'url' => "accounts/addproofer.php"];
-        $links[] = ['text' => _('Help'), 'url' => get_faq_url("faq_central.php")];
         echo headerbar_text_array($links);
         echo "</div>\n";
     } else {
         //Code to display if the user is logged in
 
         // left side
         echo "<span id='navbar-left'>";
-        echo headerbar_text_array([['text' => $site_abbreviation, 'url' => "default.php"]]);
-
-        echo " | ";
 
         $links = [];
         $links[] = ['text' => _("Activity Hub"), 'url' => "activity_hub.php"];
 
-        // use white-space: nowrap; to keep the rounds within []s together
-        $activities_string = "<span style='white-space: nowrap;'>[ ";
-        if (get_pages_proofed_maybe_simulated() < 100) {
-            $activities_string .= "<span class='text'>" . _("Activities") . ": </span>";
+        // We create 2 different ways of showing the activities, one with them
+        // all out in one string and one with with them collapsed behind a menu.
+        // We show/hide them based on the user's navbar_activity_menu preference.
+        //   0 - always collapse
+        //   1 - automatically collapse
+        //   2 - never collapse
+        // Furthermore, never collapse the menu for new users or users with
+        // fewer than 4 activities.
+        $activity_links = get_activity_links();
+
+        $is_new_user = get_pages_proofed_maybe_simulated() < 100;
+
+        // Calculate style-level CSS that will override the classes
+        // 2 - never collapse
+        if ($user->navbar_activity_menu == 2 || $is_new_user || count($activity_links) <= 3) {
+            $string_style = "style='display: inline;'";
```

Comment:
You can drop all the trailing semicolons in your HTML `style='foo: bar;'` attributes.

Reply:
Noted, but I think I'll leave them (makes it easier to add new ones if necessary, move it to stylesheet if necessary, and I'm lazy 😁 ).